src/main/java/com/fusionflux/portalcubed/blocks/blockentities/FloorButtonBlockEntity.java

```diff
@@ -43,21 +43,24 @@ public static void tick1(World world, BlockPos pos, BlockState state, FloorButto
 
             boolean isPowered = false;
             for (LivingEntity living : entities) {
-                if (living instanceof PlayerEntity || living instanceof StorageCubeEntity || living instanceof Portal1CompanionCubeEntity || living instanceof Portal1StorageCubeEntity || living instanceof OldApCubeEntity) {
-                    isPowered = true;
-                    if (state.getBlock().equals(PortalCubedBlocks.FLOOR_BUTTON)) {
-                        if (living instanceof StorageCubeEntity cube) {
-                            cube.setButtonTimer(1);
-                        }
-                        if (living instanceof CompanionCubeEntity cube) {
-                            cube.setButtonTimer(1);
+                if (living instanceof PlayerEntity || living instanceof StorageCubeEntity || living instanceof CompanionCubeEntity || living instanceof Portal1CompanionCubeEntity || living instanceof Portal1StorageCubeEntity || living instanceof OldApCubeEntity || living instanceof RedirectionCubeEntity) {
```

Comment:
`living instanceof CompanionCubeEntity` and `living instanceof RedirectionCubeEntity` are covered by `living instanceof StorageCubeEntity` because they're subclasses.
```suggestion
                if (living instanceof PlayerEntity || living instanceof StorageCubeEntity || living instanceof Portal1CompanionCubeEntity || living instanceof Portal1StorageCubeEntity || living instanceof OldApCubeEntity) {
```